src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -50,6 +50,12 @@ internal interface IDiagnosticAnalyzerService
         /// </summary>
         Task ForceAnalyzeAsync(Solution solution, Action<Project> onProjectAnalyzed, ProjectId? projectId, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// Returns true if the given <paramref name="projectId"/> was force analyzed by invoking
+        /// <see cref="ForceAnalyzeAsync(Solution, Action{Project}, ProjectId?, CancellationToken)"/> on the project or the solution.
+        /// </summary>
+        bool WasForceAnalyzed(ProjectId projectId);
```

Comment:
i somehwat hate this.  because it implies a statefulness to this service that seems very risky.